hikari/internal/data_binding.py

```diff
@@ -75,62 +75,87 @@
 Stringish = typing.Union[str, int, bool, undefined.UndefinedType, None, snowflakes.Unique]
 """Type hint for any valid that can be put in a StringMapBuilder"""
 
+JSONEncoder = typing.Callable[[typing.Union[JSONArray, JSONObject]], bytes]
+"""Type hint for hikari-compatible JSON encoders.
+
+A hikari-compatible JSON encoder is one which will take in a JSON-ish object and output either `str`
+or `bytes`.
+"""
+
+JSONDecoder = typing.Callable[[typing.Union[str, bytes]], typing.Union[JSONArray, JSONObject]]
+"""Type hint for hikari-compatible JSON decoder.
+
+A hikari-compatible JSON decoder is one which will take either a `str` or `bytes` and outputs
+the JSON-ish object, as well as raises a `ValueError` on an incorrect JSON payload being passed in.
+"""
+
 _StringMapBuilderArg = typing.Union[
     typing.Mapping[str, str], multidict.MultiMapping[str], typing.Iterable[typing.Tuple[str, str]]
 ]
 
 _APPLICATION_OCTET_STREAM: typing.Final[str] = "application/octet-stream"
+_JSON_CONTENT_TYPE: typing.Final[str] = "application/json"
+_UTF_8: typing.Final[str] = "utf-8"
 
-if typing.TYPE_CHECKING:
-    JSONDecodeError: typing.Type[Exception] = Exception
-    """Exception raised when loading an invalid JSON string."""
+default_json_dumps: JSONEncoder
+"""Default json encoder to use."""
 
-    def dump_json(_: typing.Union[JSONArray, JSONObject], /, *, indent: int = ...) -> str:
-        """Convert a Python type to a JSON string."""
-        raise NotImplementedError
+default_json_loads: JSONDecoder
+"""Default json decoder to use."""
 
-    def load_json(_: typing.AnyStr, /) -> typing.Union[JSONArray, JSONObject]:
-        """Convert a JSON string to a Python type."""
-        raise NotImplementedError
+try:
+    import orjson
 
-else:
+    default_json_dumps = orjson.dumps
+    default_json_loads = orjson.loads
+except ModuleNotFoundError:
     import json
 
-    dump_json = json.dumps
-    """Convert a Python type to a JSON string."""
+    def default_json_dumps(obj: typing.Union[JSONArray, JSONObject]) -> bytes:
+        """Encode a JSON object to a `str`."""
+        return json.dumps(obj, separators=(",", ":")).encode()
```

Comment:
```suggestion
        return json.dumps(obj, separators=(",", ":")).encode(_UTF_8)
```

Reply:
Also, consider extracting the separators out of there to not keep building the tuple constantly